lib/spack/llnl/util/tty/log.py

```diff
@@ -725,7 +899,8 @@ def _writer_daemon(stdin_multiprocess_fd, read_multiprocess_fd, write_fd, echo,
     # write_fd to terminate the reading loop, so we close the file descriptor
     # here. Forking is the process spawning method everywhere except Mac OS
     # for Python >= 3.8 and on Windows
-    if sys.version_info < (3, 8) or sys.platform != 'darwin':
+    if sys.version_info < (3, 8) \
+            or sys.platform not in ['darwin', 'cygwin']:
```

Comment:
To be clear, the way this is written, this will also skip for Cygwin on Python < 3.8. My understanding is that Windows Python before 3.8 would also require this. The if statement should trigger if:

* Python < 3.8 on MacOS
* Windows

Should this also trigger for "win32"? Should there be a function to check if sys.platform is one of "cygwin"/"win32"?

Reply:
We don't officially support cygwin, and generally don't handle the case where cygwin is used as anything but a shell around a windows compiler environment (and really that is not heavily tested). IMO this check can be completely reverted, but I'll check with the original author to be sure.

Reply:
I don't think we want to explicitly avoid the Cygwin support, but a warning without failing and a note in the getting started portion of the readme feel sufficient to me.